cdap-api/src/main/java/co/cask/cdap/api/schedule/StreamSizeSchedule.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api.schedule;
+
+/**
+ * Defines a schedule based on data availability in a stream for running a program.
+ */
+public class StreamSizeSchedule implements NotificationSchedule {
+
+  private final String name;
+
+  private final String description;
+
+  private final String sourceNamespaceId;
+
+  private final String sourceName;
+
+  private final int dataTriggerMB;
+
+  public StreamSizeSchedule(String name, String description, String sourceNamespaceId, String sourceName,
```

Comment:
So, do sourceNamespaceId and sourceName together form a stream id? Or a feed id?